power_grid_model_io_native_c/power_grid_model_io_native_c/src/vnf_pgm_converter.cpp

```diff
@@ -14,19 +14,16 @@
 
 using power_grid_model::ConstDataset;
 
-// TODO(Laurynas-Jagutis) add call_with_catch for these functions
+// TODO(Laurynas-Jagutis) add call_with_catch for these functions.
 PGM_IO_VnfConverter* PGM_VNF_create_converter(const PGM_IO_Handle* /*handle*/, char* file_buffer) {
     auto* converter = new PgmVnfConverter(file_buffer);
     parse_vnf_file_wrapper(converter);
     return reinterpret_cast<PGM_IO_VnfConverter*>(converter);
 }
 
-PGM_IO_ConstDataset const* PGM_VNF_get_input_data(const PGM_IO_Handle* /*handle*/, PGM_IO_VnfConverter* converter_ptr,
-                                                  PGM_IO_ConstDataset const* dataset) {
+char const* PGM_VNF_get_input_data(const PGM_IO_Handle* /*handle*/, PGM_IO_VnfConverter* converter_ptr) {
     auto* converter = reinterpret_cast<PgmVnfConverter*>(converter_ptr);
```

Comment:
you shouldn't have to `reinterpret_cast`. in the PGM we also use opaque pointers (forward-declared in the `.h` files, instantiated using `using ... = ...;` in the implementation. please cross-check with e.g. `PGM_PowerGridModel` whether and why you need this.